test/gtest/ucp/test_ucp_tag_xfer.cc

```diff
@@ -244,19 +249,26 @@ void test_ucp_tag_xfer::test_xfer_generic(size_t size, bool expected, bool sync)
     dt_gen_start_count  = 0;
     dt_gen_finish_count = 0;
 
+    if ((truncated) && (!count)) {
+        truncated = false;
```

Comment:
I see. So in general, it's a bad practice to change function arguments, since it may lead to all sorts of confusion. It would be better to change the 'if' below to:
`if (!truncated || (count==0)) {`